### PR TITLE
Extract MergeFilters to iterative optimizer

### PR DESCRIPTION
**Pull request by pnowojski**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java [hunk 1]
```diff
@@ -111,27 +114,33 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
...
+                                        new PushLimitThroughProject(),
+                                        new MergeLimits(),
+                                        new MergeLimitWithSort(),
+                                        new MergeLimitWithTopN(),
+                                        new PushLimitThroughMarkDistinct(),
+                                        new PushLimitThroughSemiJoin(),
+                                        new MergeLimitWithDistinct(),
+
+                                        new PruneValuesColumns(),
+                                        new PruneTableScanColumns())).build()
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java]: put `build()` in  separate line

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPredicatePushDown.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.filter;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.values;
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.expression;
+
+public class TestPredicatePushDown
+{
+    private final RuleTester tester = new RuleTester();
+    private final Rule mergeFilter = new MergeFilters();
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPredicatePushDown.java]: I would inline these fields

File: presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPredicatePushDown.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Test
+    public void testMergeFilter()
```
Comment (kokosing) [inline presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPredicatePushDown.java]: add a 'stupid' test that rule does not apply to some 'stupid' plan like `values`

**Comment by pnowojski** [comment]
Superseded by https://github.com/prestodb/presto/pull/7377

**Pull request closed without merging**